[CMake][WPE][GTK] Vendor unifdef in ThirdParty

philn opened the pull request:
#### 665e2c33e5f3739431427bc6fa389c64d3c91df1
<pre>
[CMake][WPE][GTK] Vendor unifdef in ThirdParty
<a href="https://bugs.webkit.org/show_bug.cgi?id=279711">https://bugs.webkit.org/show_bug.cgi?id=279711</a>

Reviewed by Adrian Perez de Castro.

Unifdef is not included the GNOME SDK and might not be packaged on some Linux distributions either,
so add a new CMake USE_SYSTEM_UNIFDEF option, set to ON by default. When the option is disabled the
unifdef version vendored in ThirdParty/ will be used.

* Source/CMakeLists.txt:
* Source/ThirdParty/unifdef/CMakeLists.txt: Added.
* Source/ThirdParty/unifdef/COPYING: Added.
* Source/ThirdParty/unifdef/README.WebKit: Added.
* Source/ThirdParty/unifdef/unifdef.c: Added.
* Source/ThirdParty/unifdef/unifdef.h: Added.
* Source/ThirdParty/unifdef/version.h: Added.
* Source/cmake/GLibMacros.cmake:
* Source/cmake/OptionsGTK.cmake:
* Source/cmake/OptionsWPE.cmake:

Canonical link: <a href="https://commits.webkit.org/283695@main">https://commits.webkit.org/283695@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/446ee2a5c2289846406299cb58b0cec0ecb4f1c1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/67090 "1249 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/46469 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/19722 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/71125 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/18223 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/69208 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/54267 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/18014 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/53806 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/12264 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/70157 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/42743 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/58054 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/34325 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/39415 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/15448 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/16577 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/60205 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/61374 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/15789 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/72827 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/66335 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/11049 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/15141 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/61275 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/11082 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/58113 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/61352 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14821 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/9087 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/2690 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/88103 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/42274 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/15513 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/43351 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/44537 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/43092 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc commented on Source/ThirdParty/unifdef/README.WebKit:
```diff
@@ -0,0 +1,5 @@
+https://dotat.at/prog/unifdef/unifdef-2.12.tar.gz
+
+When updating to a new release please keep this README.WebKit (updated) and CMakeLists.txt files
+intact.
```
I am having trouble parsing this... Are they to be updated or to be kept intact? How about “keep this README.WebKit and CMakeLists.txt files in place, updated as needed”?

aperezdc reviewed (approved):
LGTM with a small nit, see below.

philn committed:
[CMake][WPE][GTK] Vendor unifdef in ThirdParty
https://bugs.webkit.org/show_bug.cgi?id=279711

Reviewed by Adrian Perez de Castro.

Unifdef is not included the GNOME SDK and might not be packaged on some Linux distributions either,
so add a new CMake USE_SYSTEM_UNIFDEF option, set to ON by default. When the option is disabled the
unifdef version vendored in ThirdParty/ will be used.

* Source/CMakeLists.txt:
* Source/ThirdParty/unifdef/CMakeLists.txt: Added.
* Source/ThirdParty/unifdef/COPYING: Added.
* Source/ThirdParty/unifdef/README.WebKit: Added.
* Source/ThirdParty/unifdef/unifdef.c: Added.
* Source/ThirdParty/unifdef/unifdef.h: Added.
* Source/ThirdParty/unifdef/version.h: Added.
* Source/cmake/GLibMacros.cmake:
* Source/cmake/OptionsGTK.cmake:
* Source/cmake/OptionsWPE.cmake:

Canonical link: https://commits.webkit.org/283695@main
